server/controllers/gift.controller.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+const DB = require('./../../config/db');
+const Gift = DB.Gift;
+const Event = DB.Event;
+const Guest = DB.Guest;
+const { messages } = require('./../helper');
+
+const validateGift = (entity, eventId) => {
+  return entity && entity.dataValues.event_id === Number(eventId);
+};
+
+const isEventOwner = (eventId, userId, entity) => {
+  return Event.findOne({
+    where: {id: Number(eventId),
+      $and: {owner: userId}
+    }
+  })
+  .then(result => result && Number(result.dataValues.id) && validateGift(entity, eventId))
+  .catch(error => error);
+};
+
+const isEventGuest = (eventId, userId, entity) => {
+  return Guest.findOne({
+    where: {event_id: Number(eventId),
+      $and: {user_id: userId}
+    }
+  })
+  .then(result => result && Number(result.dataValues.id) && validateGift(entity, eventId))
+  .catch(error => error);
+};
+
+const eventIsDraft = eventId => {
+  return Event.findOne({
+    where: {id: Number(eventId),
+      $and: {status_event_id: 1}
+    }
+  })
+  .then(result => result && Number(result.dataValues.id))
+  .catch(error => error);
+};
+
+module.exports = {
+  create(req, res) {
+    let giftParams = Object.assign({}, req.body, {event_id: req.params.id});
+    Gift.create(giftParams)
+    .then(gift => isEventOwner(req.params.id, req.decoded.id, gift)
```

Comment:
without this check any logged in user can create any gift in any event behind the UI